### PR TITLE
Feature/julian 10 20

### PR DESCRIPTION
**Pull request by jcoutu**
My woefully late blog post. Please let me know if it's horrible doesn't make sense or has other issues.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/wombat-security/wombat-security.github.io/pull/33" target="_blank">ReviewNinja

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: The Grapes On Rack
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: Did you mean Grapes Of Rack? not On?

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+---
+###Grape, Rack and assorted other things to build an API.
+
+##How We Got There
+
+An API has been in discussion for a while now but recently the needs have grown stronger and the desire to create something to get more separation has been greater then ever. So after much discussion about microservices and how we would like to go about structuring our api servers it was decided to dive in with the user management and authentication first for some proof of concept. So the search for the right framework began. 
+
+_Ruby on Rails_
+
+We could use rails. We are already familiar with it (our main user interface is built on rails) but it just felt to heavy. I don't mean response times or speed I will leave that to many other people on the Internet to work out I mean heavy code wise. We don't need activerecord (we use [Sequel](http://sequel.jeremyevans.net) in other places and some of us have taken a liking to it.) We don't need an asset pipeline as we won't be returning anything except data (mostly json). We don't really need activesupport. We just need something light allowing use to do some quick queries and return back some datasets.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: Suggested edits:
* "felt too heavy"
* add commas to this sentence: "I don't mean response times or speed, I will leave that to many other people on the Internet to work out, I mean heavy code wise."
* if you care to capitalize Rails, ActiveRecord and ActiveSupport

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+_Elixir_ 
+
+Elixir specifically using the Pheonix framework were a compelling option. I have heard great things from other developers and the little bit of time I have spent playing around in it was good. It looks like a really great tool set and I would love to use it. But this is the real world where we get to think about things like ROI and long term maintenance costs. Developer hiring/training and a myriad of other reasons. This might not be the case in the future but right now where we stand it just didn't feel right. It felt like we where using a new language because we wanted to learn a new language not because it was right for the business.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: "was a compelling option"

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+_GO, Rust, and all the others we would love to try_
+
+See the last two sentences for Elixir above.
+
+_What's left_
+
+Okay so we want a lightweight ruby framework for serving apis. Well there are a bunch of options and there is always just rolling your own right. After searching high and low I kept ending up looking at [Grape](https://github.com/ruby-grape/grape). It's designed as a light-weight REST-like (their term not mine) API micro-framework for rack. Okay this seems promising what else. Well it has some basic routing some, helpers, parameter validation, and it seems to be well supported (The last commit was just 6 days ago) with a growing set of plug ins and helpers. Great let's try that.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: * remove comma between "some, helpers"

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+##So Here We Are Now What
+
+Well now that we had an idea of what framework it was time to get to work. After working on the code some and getting our normal setup, testing ci server, database, etc. in place I was left with one more major stumbling block. That was what's the best way to formt and serialize this data.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: * missing a letter in "format"

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+##So Here We Are Now What
+
+Well now that we had an idea of what framework it was time to get to work. After working on the code some and getting our normal setup, testing ci server, database, etc. in place I was left with one more major stumbling block. That was what's the best way to formt and serialize this data. 
+
+__Formatting and Serializing__
+
+While looking through our format options I just kept getting a nagging feeling that just passing back objects or collections of objects was going to bite us later on when writing the consumer. I wanted more structure but this being a primarily json api what could I do. I mean it's json and that's pretty much follow the format put your stuff in and if it passes a linter it's good to go. Then during some head banging about this I started look at the format specs out there.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: * suggested punctuation between format and put: "I follow the format, put your stuff in and if it"

File: _posts/2015-10-28-the-grapes-on-rack.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+_[JSend](http://labs.omniti.com/labs/jsend)_
+
+As I began looking at JSend first I liked it. But as I started using it some I found that it really wasn't adding much over what I was already doing on my own. It doesn't really help define collections vs object or relationships. It really was just a clean way of presenting success/failure and messages along with the failures. It ended up being a little to lightweight for me.
```
Comment (laurawhalin) [inline _posts/2015-10-28-the-grapes-on-rack.md]: * "little too lightweight"

**Pull request closed without merging**